src/components/RadioButtonGroup/styled/buttonItem.js

```diff
@@ -32,23 +44,35 @@ const StyledButtonItem = styled.span`
     ${props =>
         props.variant === 'inverse' &&
         `
-            &:hover {
-                background-color: rgba(0, 0, 0, 0.95);
-                cursor: pointer;
-            }
-        `};
-    ${props =>
-        props.isChecked &&
-        `
-            transition: transform 600ms cubic-bezier(0.02, 0.94, 0.09, 0.97), all 300ms cubic-bezier(0.17, 0.67, 0.14, 1.03);
-            transform: translate3d(0, 0, 0);
-            height: 100%;
+                &:hover {
+                    background-color: rgba(0, 0, 0, 0.95);
+                    cursor: pointer;
+                }
+            `};
+    ${props => {
+        const brandMainContrastText = props.getContrastText(props.brandMainColor);
+        const brandDarkContrastText = props.getContrastText(props.brandDarkColor);
 
-            &:hover {
-                background-color: transparent;
-                cursor: pointer;
-            }
-        `};
+        return (
+            props.isChecked &&
+            `
+                transition: transform 600ms cubic-bezier(0.02, 0.94, 0.09, 0.97), all 300ms cubic-bezier(0.17, 0.67, 0.14, 1.03);
+                transform: translate3d(0, 0, 0);
+                height: 100%;
+
+                background-color: ${props.brandMainColor};
+                border: 1px solid ${props.brandMainColor};
+                color: ${brandMainContrastText};
+
+                &:hover {
+                    background-color: ${props.brandDarkColor};
```

Comment:
We should not add `hover` styles when the element `isChecked` because it can be confused for the user